[Refactor] iconHeader && Important && Note && NoteSplit && Tip && Warning CSS refactor

imjuangarcia opened the pull request:
This PR refactors a bunch of components that are fairly similar amongst them, and removes an almost duplicate `note` component.

tvst commented:
These icons aren't quite centered:

![image](https://user-images.githubusercontent.com/690814/153731727-4719a35a-fef5-4068-a063-04ae2df423e4.png)

Here's the original, for reference:

![image](https://user-images.githubusercontent.com/690814/153731714-377b804c-96f3-4235-a54f-f18a6ed8d084.png)

tvst reviewed (approved):
The line height in these code boxes is larger in the PR than today. (2em vs 1.75em)

Do the docs use 1.75em as the line-height in general? If so, we should change the Tailwind config to make regular text have 1.75 height

imjuangarcia commented:
> These icons aren't quite centered

Fixed on [ad2020c](https://github.com/streamlit/docs/pull/287/commits/ad2020c0e957b3dd29e9368d50ae1b925f30f038)

> The line height in these code boxes is larger in the PR than today. (2em vs 1.75em)

Fixed on [f961d47](https://github.com/streamlit/docs/pull/287/commits/f961d478648e75d927367472acb48d08ffce434b)

> Do the docs use 1.75em as the line-height in general? If so, we should change the Tailwind config to make regular text have 1.75 height.

I agree! I think it'd be good to make changes like these, that affect multiple components, when we remove the global styles, with all the PRs merged, to ensure we don't accidentally break anything. I'll try it out after I merge everything!
